### DISCUSSION TITLE
How to disable AMD in Webpack 5?

### DISCUSSION THREAD

**Discussion by vxncetxn** [webpack/webpack#14346] [Q&A]
A particular library prescribes that we add this to the webpack config file for it to work:

`
module: {
  rules: [
    {
      parser: {
        amd: false
      }
    }
  ]
}
`

which works fine in Webpack 4 but not in Webpack 5, giving this error: `ModuleNotFoundError: Module not found: ValidationError: Invalid parser object. Json Modules Plugin has been initialized using a parser object that does not match the API schema.  - parser has an unknown property 'amd'. These properties are valid:
   object { parse? }`. 

May I know what is the Webpack 5 equivalent of writing the above lines of code to disable AMD? Thanks!

**Comment by alexander-akait** [comment]
Please use https://webpack.js.org/configuration/other-options/#amd

**Reply by SharperImage** [reply]
How do we use this property? Is it like:
`amd: {
   package_name: false,
}`?

The documentation is pretty incomplete.

I attempted the above and it did not work.

**Reply by Fitz6** [reply]
@SharperImage I agree with your point that the documentation is incomplete. I tried to make sense of it and found that the default value for this option is an empty object, which will be assigned to `define.amd` at runtime. Most AMD modules are loaded by checking `typeof define === "function" && define.amd`. Therefore, you should configure it as `amd: false` to disable it.

@alexander-akait Perhaps the default value of `amd: false` would be more appropriate, as AMD support may no longer need to be enabled by default nowadays.

**Reply by alexander-akait** [reply]
@Fitz6 Yes, you are right, but we can't do it right now, because it is a breaking change, only for webpack@6, we already have plans to disable it
